ParametricObjectCreation/HoneycombSolid/HoneycombSolid.py

```diff
@@ -0,0 +1,235 @@
+		polygon = Part.makePolygon(figure)
+
+		# move it to the center of the container box	
+		polygon.translate(Base.Vector(length/2, width/2, 0))
+
+		# create a face for the first polygon
+		f1=Part.Face([polygon])
+
+		#######################################################################
+		# create copies of poligon using radial pattern
+
+		# calculate how many circunferencias needs to cover the maximun length of the container box
+		n_cols = math.ceil( (length / (radius + tickness) / 2) )
+		n_rows = math.ceil( (width / (radius + tickness) / 2) + 3)
+		# FreeCAD.Console.PrintMessage("n_cols: " + str(n_cols) + " n_rows: " + str(n_rows) + "\n")
+	
+		# To store all the poligons face
+		e_faces = []
+		e_faces.append(f1) # add the first one created before
+
+		# Iterate over each imaginary circle which circunference contains the center of the polygon circle 
+		for column in range(-n_cols, n_cols):
+			for row in range(-n_rows, n_rows):
+				# omit the first
+				if (column != 0) or (row != 0):
+					if column % 2 != 0:
+					# odd
+						if row % 2 != 0:
+						# odd
+							centers_distance = tickness + 2 * radius * math.sin(edges_angle)
+							x_delta = centers_distance *  math.sin(edges_angle)
+							y_delta = centers_distance * math.cos(edges_angle)
+
+							x_origin = column * x_delta 
+							y_origin = row * y_delta
+		
+							delta_x_y = Base.Vector(x_origin, y_origin,0)
+		
+							polygon_copy = polygon.copy()
+							polygon_copy.translate(delta_x_y)		
+							fn=Part.Face([polygon_copy])
+							e_faces.append(fn)
+
+					else:
+					# even
+						if row <= (n_rows/2) and row >= (-n_rows/2):
+							centers_distance = tickness + 2 * radius * math.sin(edges_angle)
+							x_delta = centers_distance  * math.sin(edges_angle)
+							y_delta = centers_distance
+							
+							x_origin = column * x_delta 
+							y_origin = row * y_delta
+		
+							delta_x_y = Base.Vector(x_origin, y_origin,0)
+		
+							polygon_copy = polygon.copy()
+							polygon_copy.translate(delta_x_y)		
+							fn=Part.Face([polygon_copy])
+							e_faces.append(fn)
+
+		# join all the faces
+		mshell = Part.makeShell(e_faces)
+		extruded_polygon = mshell.extrude(Base.Vector(0,0,height))
+
+		# cut the array of solids using the container box
+		shape = container.cut(extruded_polygon) # comment it to see the array of solids
+
+		fp.Shape = shape # comment it to see the array of solids
+		#fp.Shape = extruded_polygon # uncomment it to see the array of solids
+
+	def __getstate__(self):
+		''' When saving the document this object gets stored using Python's cPickle module.
+		Since we have some un-pickable here -- the Coin stuff -- we must define this method
+		to return a tuple of all pickable objects or None.
+		'''
+		return self.Type, self.Version
+
+	def __setstate__(self,state):
+		''' When restoring the pickled object from document we have the chance to set some
+		internals here. Since no data were pickled nothing needs to be done here.
```

Comment:
The comment states "no data were pickled". Please update.

Reply:
Done!